src/Native/Unix/System.Native/pal_process.c

```diff
@@ -357,25 +356,25 @@ static int32_t ConvertRLimitResourcesPalToPlatform(RLimitResources value)
             return RLIMIT_NOFILE;
     }
 
-    assert_msg(false, "Unknown RLIMIT value", static_cast<int>(value));
+    assert_msg(false, "Unknown RLIMIT value", (int)value);
     return -1;
 }
 
+#define LIMIT_MAX(T) _Generic(((T)0), \
+  unsigned int: UINT_MAX,             \
+  unsigned long: ULONG_MAX,           \
+  default: 0)
```

Comment:
> < sim642> Actually thinking about it now, if you don't have all int types associated there, it might be better to remove the default: 0 as well
> < sim642> Might cause weird issues if it's neither and some max turns out to be 0 which may have weird consequences instead of a compilation error

Reply:
Makes sense. Removed `default` case as we most probably want to vet all these cases and defaulting to `0` is likely going to cause or hide problems.